src/index.ts

```diff
@@ -14,255 +18,121 @@ const TrieSearch = require('trie-search');
-        usersTrie = new TrieSearch('displayName');
-        usersTrie.addAll(Object.values(displayNames));
+        let newUser = userService.createNewUser(client.id);
 
         // Update clients about new user
-        client.emit(Constants.DISPLAY_NAME, displayNames[client.id]);
-        io.emit(Constants.USERS, displayNames);
+        client.emit(Constants.DISPLAY_NAME, newUser);
+        io.emit(Constants.USERS, userService.getRandomUsers());
     });
 
     /**
      * Received when client searches for another user
      */
     client.on(Constants.SEARCH_USERS, (searchTerm: string) => {
-        client.emit(Constants.SEARCH_USERS, usersTrie.get(searchTerm));
+        client.emit(Constants.SEARCH_USERS, userService.searchUsers(searchTerm));
     });
 
     /**
      * Received when client clicks on another user and presses connect.
      */
     client.on(Constants.CREATE_ROOM, (data: Types.CreateRoom) => {
-        const roomid = uuid.v1(); // Generate random time-based id
-        const newRoom = {
-            roomid: roomid,
-            owner: client.id,
-            requestSent: false,
-            invited: data.invited,
-        };
-        newRoom.invited[client.id].accepted = true;
-
-        rooms[roomid] = newRoom;
-        client.emit(Constants.CREATE_ROOM_SUCCESS, { roomid });
-        logInfo(`${client.id} has created room with id ${roomid}`);
+        let newRoomId = roomService.createRoom(client.id, data.invited);
+        client.emit(Constants.CREATE_ROOM_SUCCESS, { roomId: newRoomId });
     });
 
     /**
      * Called when user goes directly to room link or right after owner has created room.
      */
     client.on(Constants.CONNECT_TO_ROOM, (data: Types.ConnectRoom) => {
-        // Connect to room
-        userRooms[client.id] ? userRooms[client.id].push(data.roomid) : (userRooms[client.id] = [data.roomid]);
-
-        // Update room
-        const room = rooms[data.roomid];
-        room.invited[client.id].accepted = true;
-        logInfo(`${displayNames[client.id]} connected to room ${data.roomid}`);
-        let invitedUserIds = Object.keys(room.invited);
-        invitedUserIds.forEach(socketid => {
-            if (room.invited[socketid].accepted) {
-                io.to(socketid).emit(Constants.ROOM_STATUS, {
-                    type: Constants.USER_CONNECT,
-                    roomid: data.roomid,
-                    full: invitedUserIds.every(userid => {
-                        return room.invited[userid].accepted;
-                    }),
-                    owner: room.owner,
-                    userid: client.id,
-                });
-            }
-        });
-        return;
+        roomService.connectToRoom(client.id, data.roomId)
+        // Add room to user's list of connected rooms
+        userRooms[client.id] ? userRooms[client.id].push(data.roomId) : (userRooms[client.id] = [data.roomId]);
```

Comment:
Hm, I'm usually not a fan of mutation of variables. What about changing it to an assignment instead, like
```
userRooms[client.id] = userRooms[client.id] ? [...userRooms[client.id], data.roomId] : [data.roomId];
```

Reply:
Thoughts?

Reply:
Agreed. It'd also be consistent with the way we are assigning in redux logic. Changed!